src/app/home/todos/todos.component.ts

```diff
@@ -47,28 +59,44 @@ export class TodosComponent implements OnInit, OnDestroy {
   }
 
   get filterValue(): TodoFilter {
-    if (typeof window !== 'undefined') {
-      return (window.location.hash.replace(/^#\//, '') as TodoFilter) || TodoFilter.All;
+    if (typeof window !== "undefined") {
+      return (
+        (window.location.hash.replace(/^#\//, "") as TodoFilter) ||
+        TodoFilter.All
+      );
     }
     return TodoFilter.All;
   }
 
   get itemsLeft() {
-    return (this.todos || []).filter(t => !t.completed).length;
+    return (this.todos || []).filter((t) => !t.completed).length;
   }
 
   clearCompleted() {
-    (this.todos || []).filter(t => t.completed).forEach(t => this.delete.emit(t));
+    (this.todos || [])
+      .filter((t) => t.completed)
+      .forEach((t) => this.delete.emit(t));
+
+    this.todos.splice(0, this.todos.length);
```

Comment:
Why is this needed?

Reply:
The added items were not deleted, so I added.